### ISSUE TITLE
Do not execute build when the application is being closed

### ISSUE THREAD

**Issue by qtran-n4** [eclipse/n4js#262]
- Open N4JS IDE with a project

- Clean build the project. In the middle of the build, cancel the build

- Close N4JS IDE
In the console log

```
!MESSAGE org.eclipse.n4js.generator.SubGeneratorType  - org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator cannot be found.

66759 [Worker-5] WARN  org.eclipse.n4js.generator.SubGeneratorType  - org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator cannot be found.

!ENTRY org.apache.log4j 2 0 2017-09-27 17:48:36.253
!MESSAGE org.eclipse.n4js.generator.SubGeneratorType  - org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator cannot be found.

66760 [Worker-5] WARN  org.eclipse.n4js.generator.SubGeneratorType  - org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator cannot be found.

!ENTRY org.apache.log4j 2 0 2017-09-27 17:48:36.254
!MESSAGE org.eclipse.n4js.generator.SubGeneratorType  - org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator cannot be found.

66761 [Worker-5] WARN  org.eclipse.n4js.generator.SubGeneratorType  - org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator cannot be found.

!ENTRY org.apache.log4j 2 0 2017-09-27 17:48:36.256
!MESSAGE org.eclipse.n4js.generator.SubGeneratorType  - org.eclipse.n4js.transpiler.es.EcmaScriptSubGenerator cannot be found.
```

The error messages are bugs but the more serious problem here is that. The build is executed even when the application is being closed. This is due the Eclipse's internal implementation of AutobuildJob

```
class AutoBuildJob ... {

@Override
	public IStatus run(IProgressMonitor monitor) {
		//synchronized in case build starts during checkCancel
		synchronized (this) {
			if (monitor.isCanceled()) {
				return canceled();
			}
		}
		//if the system is shutting down, don't build
		if (systemBundle.getState() == Bundle.STOPPING)
			return Status.OK_STATUS;
		try {
			doBuild(monitor);
			lastBuild = System.currentTimeMillis();
			//if the build was successful then it should not be recorded as interrupted
			setInterrupted(false);
			return Status.OK_STATUS;
		} catch (OperationCanceledException e) {
			return canceled();
		} catch (CoreException sig) {
			return sig.getStatus();
		}
	}

```

Note that the check
```
 		if (systemBundle.getState() == Bundle.STOPPING)
			return Status.OK_STATUS;
```
may be false even if the application is being closed. This is because the OSGI will enter the STOPPING only after the application has terminated (https://bugs.eclipse.org/bugs/show_bug.cgi?id=75772). 

Acceptance Criteria:

- Abort the build when the application is being closed
- As a side effect of the fix, the bug with these error messages is also fixed

**Comment by qtran-n4** [comment]
This is won't fix.
